lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1705,16 +1705,7 @@ getAccessorPropertyType(const clang::FunctionDecl *accessor, bool isSetter,
 /// of this class as Swift generic type params.
 static bool
 shouldSuppressGenericParamsImport(const clang::ObjCInterfaceDecl *decl) {
-  while (decl) {
-    StringRef name = decl->getName();
-    if (name == "NSArray" || name == "NSDictionary" || name == "NSSet" ||
-        name == "NSOrderedSet" || name == "NSEnumerator" ||
-        name == "NSMeasurement") {
-      return true;
-    }
-    decl = decl->getSuperClass();
-  }
-  return false;
+  return decl->hasAttr<clang::SwiftImportAsNonGenericAttr>();
```

Comment:
The existing code explicitly looked up the superclass chain. Does `clang:: SwiftImportAsNonGenericAttr` actually get inherited by subclasses or are we changing behavior here? I suspect it's the latter (which the `SBElementArray` addition implies), but that makes this a source-compatibility break for Objective-C APIs that subclass `NSMutable<something>`.

In Swift 3 compatibility mode, should we follow the superclass chain here?